numba_passthru/numba_passthru.py

```diff
@@ -90,7 +90,9 @@ def obj(self):
             return self._obj
 
         def __eq__(self, other):
-            return isinstance(other, PassThruContainer) and self.obj is other.obj
+            if not isinstance(other, PassThruContainer):
+                raise NotImplementedError
```

Comment:
Shouldn't this be `return NotImplemented`? This behaves weirdly with a type like:
```python
class AlwaysEqual:
    def __eq__(self, other):
        return True
```
```python
>>> AlwaysEqual() == PassThruContainer(...)
True
>>> PassThruContainer(...) == AlwaysEqual()
NotImplementedError
```
vs using `NotImplemented`:
```python
>>> AlwaysEqual() == PassThruContainer(...)
True
>>> PassThruContainer(...) == AlwaysEqual()
True
```